.github/workflows/python-app.yml

```diff
@@ -0,0 +1,35 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: offer-fit-integration
+
+on: push
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install poetry
+      run: pipx install poetry
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.10"
+        cache: "poetry"
+    - name: Install dependencies
+      run: |
+        poetry install
+        poetry add ruff pytest-cov
+    - name: Lint with Ruff
+      run: |
+        poetry run ruff --output-format=github .
```

Comment:
As a supplement to an inadequate compiler, running linters belongs in a `pre-receive` or `pre-commit` hook. Also, static code analysis rules need to be kept in a detailed file so that they're not susceptible to change over time.

Linting code does not make any relevant contribution to verifying that the code in a branch integrates correctly with the rest of the code base. It does ensure that static code analysis rules (which may be changed over time) are applied consistently after commits were already made. Whether this is relevant to a code base is a _very_ subjective choice. It is also just wasted effort. Besides SCA rules being susceptible to change over time, by the time the pipeline runs the code was already committed and pushed. Fixing any linter errors requires either creating a new commit or rebasing the existing commits interactively and editing the ones that don't pass static code analysis. This is impractical when compared to simply making sure every commit is verified by the linter (or, at the very least, that every push is verified).

As a side rhetorical question: what does `ruff` add to the table that `flake8` and `pylint` don't already provide? I see [it's supposed to be faster](https://pythonspeed.com/articles/pylint-flake8-ruff/). I am not impressed.